packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,381 @@
+            _nameSlot,
+            vm.load(address(token), _nameSlot)
+        );
+        vm.store(
+            Predeploys.GOVERNANCE_TOKEN,
+            _symbolSlot,
+            vm.load(address(token), _symbolSlot)
+        );
+        vm.store(
+            Predeploys.GOVERNANCE_TOKEN,
+            _ownerSlot,
+            bytes32(uint256(uint160(cfg.governanceTokenOwner())))
+        );
+
+        /// Reset so its not included state dump
+        vm.etch(address(token), "");
+        vm.resetNonce(address(token));
+    }
+
+    /// @notice This contract has no initializer.
+    function _setL1Block() internal {
+        address impl = _predeployToCodeNamespace(Predeploys.L1_BLOCK_ATTRIBUTES);
+        vm.etch(
+            impl,
+            vm.getDeployedCode("L1Block.sol:L1Block")
+        );
+
+        /// Post Checks
+        _verifyProxyImplementationAddress(Predeploys.L1_BLOCK_ATTRIBUTES, impl);
+    }
+
+    /// @dev Returns true if the adress is not proxied.
+    function _notProxied(address _addr) internal pure returns (bool) {
+        return _addr == Predeploys.GOVERNANCE_TOKEN || _addr == Predeploys.WETH9;
+    }
+
+    /// @dev Returns true if the address is a predeploy.
+    function _isDefinedPredeploy(address _addr) internal pure returns (bool) {
+        return _addr == Predeploys.L2_TO_L1_MESSAGE_PASSER || _addr == Predeploys.L2_CROSS_DOMAIN_MESSENGER
+            || _addr == Predeploys.L2_STANDARD_BRIDGE || _addr == Predeploys.L2_ERC721_BRIDGE
+            || _addr == Predeploys.SEQUENCER_FEE_WALLET || _addr == Predeploys.OPTIMISM_MINTABLE_ERC20_FACTORY
+            || _addr == Predeploys.OPTIMISM_MINTABLE_ERC721_FACTORY || _addr == Predeploys.L1_BLOCK_ATTRIBUTES
+            || _addr == Predeploys.GAS_PRICE_ORACLE || _addr == Predeploys.DEPLOYER_WHITELIST || _addr == Predeploys.WETH9
+            || _addr == Predeploys.L1_BLOCK_NUMBER || _addr == Predeploys.LEGACY_MESSAGE_PASSER
+            || _addr == Predeploys.PROXY_ADMIN || _addr == Predeploys.BASE_FEE_VAULT || _addr == Predeploys.L1_FEE_VAULT
+            || _addr == Predeploys.GOVERNANCE_TOKEN || _addr == Predeploys.SCHEMA_REGISTRY || _addr == Predeploys.EAS;
+    }
+
+    /// @dev Function to compute the expected address of the predeploy implementation
+    ///      in the genesis state.
+    function _predeployToCodeNamespace(address _addr) internal pure returns (address) {
+        return address(
+            uint160(uint256(uint160(_addr)) & 0xffff | uint256(uint160(0xc0D3C0d3C0d3C0D3c0d3C0d3c0D3C0d3c0d30000)))
+        );
+    }
+
+    function _verifyProxyImplementationAddress(address _proxy, address _impl) internal view {
+        require(
+            vm.load(_proxy, PROXY_IMPLEMENTATION_ADDRESS) == bytes32(uint256(uint160(_impl))),
+            "Expected different address at Proxys PROXY_IMPLEMENTATION_ADDRESS storage slot"
+        );
+    }
+
+    function _verifyCantReinitialize(address _contract, address _arg) internal {
+        vm.expectRevert();
```

Comment:
There isn't a good way to know if it is revering due to abi mismatch or because of "already initialized"

Reply:
[Added](https://github.com/ethereum-optimism/optimism/pull/9292/files#diff-2b05d0be8d7d8de7870ac51225a0728adeb9a6df9819eaa654c455a23b8f509eR441-R442) a comment about this

Reply:
We should get the error `Initializable: contract is already initialized`, like we do here: https://github.com/ethereum-optimism/optimism/blob/13b43958a97002b49f9388774859c6cba68ed9b8/packages/contracts-bedrock/test/vendor/Initializable.t.sol#L292

```suggestion
        vm.expectRevert("Initializable: contract is already initialized");
```

It's rare that we'll want to use `vm.expectRevert()` with no args, because that passes on any revert which is an antipattern. If you expect no revert message, use `vm.expectRevert("")` to convey that.

Reply:
Added revert string